esp8266/modesp.c

```diff
@@ -37,535 +37,57 @@
-
-    esp_socket_obj_t *s = esp_socket_make_new_base();
-    s->espconn = m_new_obj(struct espconn);
-    espconn_create(s->espconn);
-
-    s->espconn->reverse = s;
-    // TODO: UDP Support
-    s->espconn->type = ESPCONN_TCP;
-    s->espconn->state = ESPCONN_NONE;
-
-    s->espconn->proto.tcp = m_new_obj(esp_tcp);
-
-    return s;
-}
-
-// method socket.close()
-STATIC mp_obj_t esp_socket_close(mp_obj_t self_in) {
-    esp_socket_obj_t *s = self_in;
-
-    if (esp_socket_listening == s) {
-        esp_socket_listening = NULL;
-    }
-
-    if (s->espconn->state != ESPCONN_NONE && s->espconn->state != ESPCONN_CLOSE) {
-        espconn_disconnect(s->espconn);
-    }
-
-    if (s->connlist != NULL) {
-        mp_obj_list_set_len(s->connlist, 0);
-    }
-
-    return mp_const_none;
-}
-STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_socket_close_obj, esp_socket_close);
-
-// method socket.__del__()
-STATIC mp_obj_t esp_socket___del__(mp_obj_t self_in) {
-    esp_socket_obj_t *s = self_in;
-
-    esp_socket_close(self_in);
-
-    if (s->fromserver) {
-        espconn_delete(s->espconn);
-    }
-
+STATIC mp_obj_t esp_deepsleep(mp_uint_t n_args, const mp_obj_t *args) {
+    system_deep_sleep(n_args > 0 ? mp_obj_get_int(args[0]) : 0);
     return mp_const_none;
 }
-STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_socket___del___obj, esp_socket___del__);
-
-// method socket.bind(address)
-STATIC mp_obj_t esp_socket_bind(mp_obj_t self_in, mp_obj_t addr_in) {
-    esp_socket_obj_t *s = self_in;
-
-    mp_uint_t port = netutils_parse_inet_addr(addr_in,
-        s->espconn->proto.tcp->remote_ip, NETUTILS_BIG);
-    s->espconn->proto.tcp->local_port = port;
-
-    return mp_const_none;
-}
-STATIC MP_DEFINE_CONST_FUN_OBJ_2(esp_socket_bind_obj, esp_socket_bind);
-
-STATIC void esp_socket_recv_callback(void *arg, char *pdata, unsigned short len) {
-    struct espconn *conn = arg;
-    esp_socket_obj_t *s = conn->reverse;
-
-    if (s->cb_recv != mp_const_none) {
-        call_function_2_protected(s->cb_recv, s, mp_obj_new_bytes((byte *)pdata, len));
-    } else {
-        if (s->recvbuf == NULL) {
-            s->recvbuf = gc_alloc(len, false);
-            s->recvbuf_len = len;
-            if (s->recvbuf != NULL) {
-                memcpy(s->recvbuf, pdata, len);
-            }
-        } else {
-            s->recvbuf = gc_realloc(s->recvbuf, s->recvbuf_len + len);
-            if (s->recvbuf != NULL) {
-                memcpy(&s->recvbuf[s->recvbuf_len], pdata, len);
-                s->recvbuf_len += len;
-            }
-        }
-        if (s->recvbuf == NULL) {
-            esp_socket_close(s);
-            return;
-        }
-    }
-}
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(esp_deepsleep_obj, 0, 1, esp_deepsleep);
 
-STATIC void esp_socket_sent_callback(void *arg) {
-    struct espconn *conn = arg;
-    esp_socket_obj_t *s = conn->reverse;
-
-    if (s->cb_sent != mp_const_none) {
-        call_function_1_protected(s->cb_sent, s);
-    }
+STATIC mp_obj_t esp_adc() {
```

Comment:
ADC capability is provided by pyb.ADC() class.  Yes that's overkill for ESP which seems to have only 1 ADC pin, but would be best to provide some consistency here, no?